include/mmtf/structure_data.hpp

```diff
@@ -188,6 +188,13 @@ struct StructureData {
     std::vector<std::string>         chainNameList;
     std::vector<int32_t>             groupsPerChain;
     std::vector<int32_t>             chainsPerModel;
+    msgpack::zone                    msgpack_zone;
+    msgpack::object                  bondProperties;
```

Comment:
would it make sense (and would it simplify anything) to store `std::map<std::string, msgpack::object>` here? I'm thinking that instead of this:
```C++
std::map<std::string, msgpack::object> atom_data_map_in;
atom_data_map_in["256_atomColorList"] = msgpack::object(clist_out, sd.msgpack_zone);
sd.atomProperties = msgpack::object(atom_data_map_in, sd.msgpack_zone);
```
It would allow to write this:
```C++
sd.atomProperties["256_atomColorList"] = msgpack::object(clist_out, sd.msgpack_zone);
```
Or even better, we should provide a template "setter" method which hides the msgpack zone magic:
```C++
sd.setAtomProperty("256_atomColorList", clist_out);
```

Reply:
one thing we could do, is leave it as a `std::map<std::string, msgpack::object>` and have a wrapper function that turns it into a `msgpack::object` and then returns a map decoder to that msgpack object.  

In my hands, msgpack::objects can be weird, but it's a possibility.

Reply:
I see. One could also modify the `MapDecoder` to make it work with `std::map<std::string, msgpack::object>` directly. Should be trivial since all the constructor does is to translate the `msgpack::object` to a `std::map<std::string, msgpack::object*>` (`data_map_`).

To be fair I do not recall why we used an `object*` there and it might just simply have been a copy/paste from the mmtf-c code and it could be done in a more elegant way. I am more than ok with you adapting `MapDecoder` if this helps...

Does anyone know if there is anything important to consider when copying `msgpack::object` objects? Or would it be an expensive operation if we copy a `std::map<std::string, msgpack::object>` object? If that's the case, the `MapDecoder` should keep the `object*` and the documentation of the constructor should specify that the original object shall not be deleted while the `MapDecoder` object is used...